src/formatDate.js

```diff
@@ -8,7 +8,25 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateComponents = date.split(fromFormat[3]);
+  const result = [];
+
+  result[toFormat.indexOf('DD')] = dateComponents[fromFormat.indexOf('DD')];
+  result[toFormat.indexOf('MM')] = dateComponents[fromFormat.indexOf('MM')];
+
+  let year = fromFormat.includes('YYYY')
+    ? dateComponents[fromFormat.indexOf('YYYY')]
+    : dateComponents[fromFormat.indexOf('YY')];
+
+  if (!toFormat.includes('YYYY')) {
+    year = year.length > 2 ? year.slice(2, 4) : year;
+    result[toFormat.indexOf('YY')] = year;
+  } else {
+    year = year.length === 2 ? (year < 30 ? `20${year}` : `19${year}`) : year;
```

Comment:
The logic for converting a two-digit year to a four-digit year based on the value is implemented correctly. It follows the rule specified in the task description.